enhancements/ingress/set_eip_nlb_ingress.md

```diff
@@ -0,0 +1,894 @@
+4. In this case, the cluster admin must directly update
+   `spec.endpointPublishingStrategy.loadBalancer.providerParameters.aws.networkLoadBalancerParameters.eipAllocations` to the
+   current value of the `service.beta.kubernetes.io/aws-load-balancer-eip-allocations` service.
+5. The Ingress Operator will resolve the `LoadBalancerProgressing` condition back to
+   `Status: False` as long as `EIPAllocations` is equivalent to `service.beta.kubernetes.io/aws-load-balancer-eip-allocations`.
+
+#### Updating EIP in the default Classic IngressController after installation
+
+We support updating the EIP in the default Classic `IngressController`.
+
+Procedure:
+Add the `ingress.operator.openshift.io/auto-delete-load-balancer` to the `IngressController`.
+```sh
+% oc -n openshift-ingress-operator annotate ingresscontrollers/default ingress.operator.openshift.io/auto-delete-load-balancer=
+ingresscontroller.operator.openshift.io/default annotated
+```
+
+Add the following under spec:
+```yaml
+spec:
+  endpointPublishingStrategy:
+    loadBalancer:
+      dnsManagementPolicy: Managed
+      providerParameters:
+        aws:
+          networkLoadBalancer:
+            eipAllocations:
+            - eipalloc-0956fea34de4cb7ab
+            - eipalloc-0e9a3077a70de050a
+            - eipalloc-0b69fc4691f54cdd0
+            - eipalloc-01e6ba6cbba1a391b
+            - eipalloc-0e242df173f906112
+          type: NLB
+        type: AWS
+      scope: External
+    type: LoadBalancerService
+```
+
+Check if the router-default service was created with proper annotations.
+
+```yaml
+% oc get svc router-default -n openshift-ingress -o yaml         
+apiVersion: v1
+kind: Service
+metadata:
+  annotations:
+    service.beta.kubernetes.io/aws-load-balancer-eip-allocations: eipalloc-0956fea34de4cb7ab,eipalloc-0e9a3077a70de050a,eipalloc-0b69fc4691f54cdd0,eipalloc-01e6ba6cbba1a391b,eipalloc-0e242df173f906112
+...
+  type: LoadBalancer
+status:
+  loadBalancer:
+    ingress:
+    - hostname: a8491585620154d5992dcdecc06c5779-7938285b62f7bebe.elb.us-east-1.amazonaws.com
+```
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+Currently, there is no unique consideration for making this change work with Hypershift.
+
+#### Standalone Clusters
+
+This proposal does not have any special implications for standalone clusters.
+
+#### Single-node Deployments or MicroShift
+
+This proposal does not have any special implications for single-node
+deployments or MicroShift.
+
+### Implementation Details/Notes/Constraints
+
+The process in general works as follows:
+- cluster-admin sets spec.
+- desiredLoadBalancerService sets the annotation based on spec.
+- syncIngressControllerStatus sets status based on the annotation.
+
+If the cluster-admin manually sets the annotation, then in that case syncIngressControllerStatus will reflect it to status.
+More details on the implementation are as follows:
+
+#### 1. Create a new IngressController with EIPs:
```

Comment:
If this is implmentation details for a custom (not default) IngressController, you should say:
```suggestion
#### 1. When creating a customer IngressController with EIPs:
```